src/client/UrlNode.ts

```diff
@@ -0,0 +1,205 @@
+/**
```

Comment:
is it not possible to import this directly from next compiled instead of forking some router logic?

It should be clear what was changed in case we need to update it again with next.js core

Reply:
I didn't find a way. They put it in the package as compiled file with other modules.
So we can move it to a separate issue.